tests/test_controllers.py

```diff
@@ -126,6 +126,23 @@ async def test_create_adapter_controller(mocker: MockerFixture):
     assert isinstance(ctrl, OdinAdapterController)
 
 
+@pytest.mark.asyncio
+async def test_given_no_module_create_adapter_controller_succeeds(
+    mocker: MockerFixture,
+):
+    controller = OdinController(IPConnectionSettings("", 0))
+
+    controller.connection = mocker.AsyncMock()
+    controller.connection.open = mocker.MagicMock()
+
+    controller.connection.get.side_effect = [
+        {"adapters": ["adapter1"]},
+        {"": {"value": "test_module"}},
+    ]
+
+    await controller.initialise()
+
+
```

Comment:
Could you assert isinstance here to confirm the fallback sub controller was used?